downloader/src/main/kotlin/vcs/Git.kt

```diff
@@ -108,7 +108,7 @@ object Git : GitBase() {
                 File(gitInfoDir, "sparse-checkout").writeText(vcsPath)
             }
 
-            val committish = if (vcsRevision != null && vcsRevision.isNotEmpty()) vcsRevision else "HEAD"
```

Comment:
Basically, only the `else` branch in the original code is wrong. Instead of always falling back to `HEAD` we should try to guess the revision from a tag. I'd like to implement this in a more generic way across all supported VCS by requiring them to implement a `listTags()` (or similar) function that is used by a `guessRevisionFromVersion()` (or similar) base function. I'll try to come up with something today.